honeybee_radiance/modifier/material/bsdf.py

```diff
@@ -233,6 +233,30 @@ def angle_basis(self, value):
         else:
             self._angle_basis = self.find_angle_basis(self.bsdf_file)
 
+    @property
+    def sampling_type(self):
+        """Return rfluxmtx parameters sampling type based on the angle basis.
+
+        Values are:
+            * kf for klems full.
+            * kh for klems half.
+            * kq for klems quarter.
+
+        Raises:
+            ValueError: If angle basis is ``TensorTree``.
+
+        """
+        _mapper = {
+        'Klems Full': 'kf', 'Klems Half': 'kh', 'Klems Quarter': 'kq'
+        }
+        if self.angle_basis == 'TensorTree':
+            raise ValueError(
+                'There is no valid sampling type for Tensor Tree BSDFs. they can only '
+                'be used for a in-scene modifiers.'
```

Comment:
It seems a little odd to get an exception raised just for checking a property. Maybe we should just return `None` when checking the property and then, if this is illegal for a given workflow, we raise the exception in that workflow.